libp2p/security/secure_transport_interface.py

```diff
@@ -0,0 +1,27 @@
+from abc import ABC, abstractmethod
+
+# pylint: disable=W0105
+
+"""
+Transport that is used to secure a connection. This transport is
+chosen by a security transport multistream module.
+
+Relevant go repo: https://github.com/libp2p/go-conn-security/blob/master/interface.go
+"""
+class ISecureTransport(ABC):
+
+    @abstractmethod
+    async def secure_inbound(self, conn):
+        """
+        Secure the connection, either locally or by communicating with opposing node via conn,
+        for an inbound connection (i.e. we are not the initiator)
+        :return: secure connection object (that implements secure_conn_interface)
+        """
+
+    @abstractmethod
+    async def secure_outbound(self, conn, peer_id):
```

Comment:
Do we actually need the peer_id for this function? I didn't see either implementation use it.

Reply:
This was a go interface so I just followed the precedent.